lib/broadway/producer.ex

```diff
@@ -111,10 +135,12 @@ defmodule Broadway.Producer do
     case module.handle_demand(demand, module_state) do
```

Comment:
If we are over the rate limit, then we shouldn't even call `handle_demand`, as that can lead to more data to be produced, which we can't send downstream anyway. In other words, once we start rate limiting, then we should also "buffer the demand".